karabo_bridge/simulation.py

```diff
@@ -60,14 +61,15 @@ def gen_combined_detector_data(source, tid_counter, corrected=False, nsources=1)
         'timestamp.frac': frac.ljust(18, '0')  # attosecond resolution
     }
 
-    # detector random data
     if corrected:
-        global _PULSES
-        _PULSES = 31
-        global _SHAPE
-        _SHAPE = (_PULSES, _MODULES, _MOD_X, _MOD_Y)
+        pulse_count = 31   # ???
+    else:
+        pulse_count = detector_info['pulses']
+    array_shape = (pulse_count, detector_info['modules']) + detector_info['module_shape']
 
-        gain_data = np.zeros(_SHAPE, dtype=np.uint16)
+    # detector random data
+    if corrected:
+        gain_data = np.zeros(array_shape, dtype=np.uint16)
         passport = [
             'SPB_DET_AGIPD1M-1/CAL/THRESHOLDING_Q3M2',
```

Comment:
maybe these can be generated out of data source name, not to confuse people in case using LPD generated data.

Reply:
Opened #28 for that.